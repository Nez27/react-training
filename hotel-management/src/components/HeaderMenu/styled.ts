import styled from 'styled-components';

const StyledHeaderMenu = styled.ul`
  display: flex;
  gap: 10px;
`;

export { StyledHeaderMenu };
