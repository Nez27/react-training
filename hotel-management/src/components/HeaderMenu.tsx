import styled from 'styled-components';

// Components
import { IoPersonCircleOutline } from 'react-icons/io5';
import { HiOutlineLogout } from 'react-icons/hi';
import ButtonIcon from '../commons/styles/ButtonIcon';

const StyledHeaderMenu = styled.ul`
  display: flex;
  gap: 10px;
`;

const HeaderMenu = () => {
  return (
    <StyledHeaderMenu>
      <li>
        <ButtonIcon aria-label="Profile">
          <IoPersonCircleOutline />
        </ButtonIcon>
      </li>
      <li>
        <ButtonIcon aria-label="Logout">
          <HiOutlineLogout />
        </ButtonIcon>
      </li>
    </StyledHeaderMenu>
  );
};

export default HeaderMenu;
