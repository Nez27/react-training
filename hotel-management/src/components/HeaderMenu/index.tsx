// Components
import { IoPersonCircleOutline } from 'react-icons/io5';
import { HiOutlineLogout } from 'react-icons/hi';
import ButtonIcon from '@component/ButtonIcon/ButtonIcon';

// Styled
import { StyledHeaderMenu } from './styled';

const HeaderMenu = () => {
  return (
    <StyledHeaderMenu>
      <li>
        <ButtonIcon
          aria-label="Profile"
          icon={<IoPersonCircleOutline />}
          iconStyle={{ size: '23px' }}
        />
      </li>
      <li>
        <ButtonIcon
          aria-label="Logout"
          icon={<HiOutlineLogout />}
          iconStyle={{size: '23px' }}
        />
      </li>
    </StyledHeaderMenu>
  );
};

export default HeaderMenu;
