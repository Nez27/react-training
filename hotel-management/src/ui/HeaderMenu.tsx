import styled from 'styled-components';
import ButtonIcon from './ButtonIcon';
import { IoPersonCircleOutline } from 'react-icons/io5';
import { HiOutlineLogout } from 'react-icons/hi';

const StyledHeaderMenu = styled.ul`
  display: flex;
  gap: 10px;
`;

const HeaderMenu = () => {
  return (
    <StyledHeaderMenu>
      <ButtonIcon>
        <IoPersonCircleOutline />
      </ButtonIcon>
      <ButtonIcon>
        <HiOutlineLogout />
      </ButtonIcon>
    </StyledHeaderMenu>
  );
};

export default HeaderMenu;
