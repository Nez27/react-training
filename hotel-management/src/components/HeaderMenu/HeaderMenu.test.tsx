import { BrowserRouter } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render } from '@testing-library/react';

// Components
import HeaderMenu from '.';

describe('HeaderMenu', () => {
  const queryClient = new QueryClient();

  const wrapper = render(
    <QueryClientProvider client={queryClient}>
      <BrowserRouter>
        <HeaderMenu />
      </BrowserRouter>
    </QueryClientProvider>
  );

  test('Should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
