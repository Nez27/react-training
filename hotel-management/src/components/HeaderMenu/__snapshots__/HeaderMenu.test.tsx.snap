// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMenu testing snapshot render 1`] = `
<ul
  className="sc-gtsqUy gOWXQs"
>
  <li>
    <button
      className="sc-bdnyFh bxHnoO"
      iconStyle={
        {
          "size": "23px",
        }
      }
      isHaveChildren={false}
      style={
        {
          "backgroundColor": undefined,
          "color": undefined,
          "fontSize": undefined,
        }
      }
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M258.9 48C141.92 46.42 46.42 141.92 48 258.9c1.56 112.19 92.91 203.54 205.1 205.1 117 1.6 212.48-93.9 210.88-210.88C462.44 140.91 371.09 49.56 258.9 48zm126.42 327.25a4 4 0 01-6.14-.32 124.27 124.27 0 00-32.35-29.59C321.37 329 289.11 320 256 320s-65.37 9-90.83 25.34a124.24 124.24 0 00-32.35 29.58 4 4 0 01-6.14.32A175.32 175.32 0 0180 259c-1.63-97.31 78.22-178.76 175.57-179S432 158.81 432 256a175.32 175.32 0 01-46.68 119.25z"
        />
        <path
          d="M256 144c-19.72 0-37.55 7.39-50.22 20.82s-19 32-17.57 51.93C191.11 256 221.52 288 256 288s64.83-32 67.79-71.24c1.48-19.74-4.8-38.14-17.68-51.82C293.39 151.44 275.59 144 256 144z"
        />
      </svg>
       
      <span />
    </button>
  </li>
  <li>
    <button
      className="sc-bdnyFh bxHnoO"
      iconStyle={
        {
          "size": "23px",
        }
      }
      isHaveChildren={false}
      style={
        {
          "backgroundColor": undefined,
          "color": undefined,
          "fontSize": undefined,
        }
      }
    >
      <svg
        aria-hidden="true"
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeWidth="2"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17 16l4-4m0 0l-4-4m4 4H7m6 4v1a3 3 0 01-3 3H6a3 3 0 01-3-3V7a3 3 0 013-3h4a3 3 0 013 3v1"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
      </svg>
       
      <span />
    </button>
  </li>
</ul>
`;
